clients/unshaded/src/main/java/tachyon/client/lineage/LineageMasterClient.java

```diff
@@ -61,6 +59,17 @@ protected String getServiceName() {
   @Override
   protected void afterConnect() {
     mClient = new LineageMasterService.Client(mProtocol);
+    if (mServiceVersion == Constants.UNKNOWN_SERVICE_VERSION) {
+      try {
+        mServiceVersion = mClient.getServiceVersion();
+      } catch (TException e) {
+        throw new RuntimeException(e.getMessage());
+      }
+      if (!mCompatibleVersions.contains(mServiceVersion)) {
+        throw new RuntimeException(ExceptionMessage.INCOMPATIBLE_VERSION.getMessage(
+            Constants.LINEAGE_MASTER_SERVICE_VERSION, mServiceVersion));
+      }
```

Comment:
I thought of that, but the mClient member has different types for different subclasses of MasterClientBase. If you can think of how to do this, let me know.